SchedulerJobs/SchedulerJobs.Services/AnonymiseHearingsConferencesDataService.cs

```diff
@@ -12,52 +15,90 @@ public interface IAnonymiseHearingsConferencesDataService
+                    { HearingIds = anonymisationData.HearingIds });
+                
+                await _videoApiClient.AnonymiseQuickLinkParticipantWithHearingIdsAsync(
+                    new AnonymiseQuickLinkParticipantWithHearingIdsRequest { HearingIds = anonymisationData.HearingIds });
+
+                await _bookingsApiClient.AnonymiseParticipantAndCaseByHearingIdAsync("hearingIds",
+                    anonymisationData.HearingIds);
+            }
 
-        private async Task DeleteUserAsync()
-        {
-            // delete users from AAD.
-            // get users that do not have hearings in the future and have had hearing more than 3 months in the past. 
-            // (exclude judges, vhos, test users, performance test users.
-            var usersToDelete = await _bookingsApiClient.GetPersonByClosedHearingsAsync();
 
-            if (usersToDelete != null && usersToDelete.Usernames != null)
+            if (anonymisationData.Usernames == null) return;
+            
+            foreach (var username in anonymisationData.Usernames)
             {
-                foreach (var username in usersToDelete.Usernames)
-                {
+                var userProfile = await _userApiClient.GetUserByAdUserNameAsync(username);
+
+                if (RemoveUserFromAd(userProfile))
                     try
                     {
-                        var user = await _userApiClient.GetUserByAdUserNameAsync(username);
-                        if (user.UserRole == AzureAdUserRoles.VhOfficer || user.UserRole == AzureAdUserRoles.StaffMember || user.IsUserAdmin)
-                            continue;
                         await _userApiClient.DeleteUserAsync(username);
                     }
                     catch (UserApiException exception)
                     {
-                        if (exception.StatusCode != (int)HttpStatusCode.NotFound && exception.StatusCode != (int)HttpStatusCode.Forbidden)
-                        {
-                            throw;
-                        }
+                        _logger.LogError(exception, ProcessingUsernameExceptionMessage,
+                            username);
                     }
+
+                try
+                {
+                    await _videoApiClient.AnonymiseParticipantWithUsernameAsync(username);
+                }
+                catch (VideoApiException exception)
+                {
+                    _logger.LogError(exception, ProcessingUsernameExceptionMessage,
+                        username);
+                }
+
+                try
+                {
+                    await _bookingsApiClient.AnonymisePersonWithUsernameForExpiredHearingsAsync(username);
+                }
+                catch (BookingsApiException exception)
+                {
+                    _logger.LogError(exception, ProcessingUsernameExceptionMessage,
+                        username);
                 }
```

Comment:
As the 'Catch' logic is the same in each, we could maybe tidy it up a bit reduce the duplication and combine the try's, with something like this: 
```
`                try
                {
                    await _videoApiClient.AnonymiseParticipantWithUsernameAsync(username);
                    await _bookingsApiClient.AnonymisePersonWithUsernameForExpiredHearingsAsync(username);
                }
                catch (Exception ex) when (ex is BookingsApiException || ex is VideoApiException)
                {
                    _logger.LogError(ex, ProcessingUsernameExceptionMessage,
                        username);
                }`
```
                
  Pretty sure that syntax is fine, but you may need to double check

Reply:
I did consider this, however the problem with this is, if an exception gets thrown at video api, it wont make the request to bookings api anymore, thats why I left it as seperate try catch instead